src/Icicle/Sea/FromAvalanche/Program.hs

```diff
@@ -245,8 +245,39 @@ seaOfXValue v
      -- TODO C escapes /= Haskell escapes
      VString x     -> text (show x)
 
+     VArray vs
+      | ArrayT t' <- t
+      -> let len = length vs
+             writes arr (v',i)
+              = prim (PrimUpdate $ PrimUpdateArrayPut t')
+                     [arr, int i, seaOfXValue v' t']
+             init
+              = prim (PrimUnsafe $ PrimUnsafeArrayCreate t')
+                     [int len]
+        in  foldl writes init (vs `List.zip` [0..])
```

Comment:
I don't think it will be required for that actually - this is only for compiletime constants